subprojects/internal-android-performance-testing/src/main/java/org/gradle/performance/android/SyncAction.java

```diff
@@ -56,4 +59,13 @@ public static void withProjectConnection(ProjectConnection connect, Action<? sup
         syncTimer.stop();
         System.out.println("Sync took " + syncTimer.duration());
     }
+
+    private static ProgressListener noOpListener() {
+        return new ProgressListener() {
+            @Override
+            public void statusChanged(ProgressEvent event) {
+                // ProgressEvents are not built using reflective proxies so we don't need to inspect them
```

Comment:
It's probably easier to understand for others unfamiliar with the TAPI if we say "Progress events have no expensive logic of their own, so we don't do anything with them. We only test the overhead of sending/receiving them"